From ea4eaa25fadc73b7ddb90957b0c3faf36a9a7886 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:41:22 -0400
Subject: [alsa] sysfs: create __ATTR_WO()

Message-id: <140972823573-424-patchset-jkysela@redhat.com>
Patchwork-id: 91120
O-Subject: [RHEL 7.1 PATCH 423/442] ALSA: sysfs: create __ATTR_WO()
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

This creates the macro __ATTR_WO() for write-only attributes, instead of
having to "open define" them.

Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

(picked from kernel: a65fcce75a75c0d41b938f86d09d42b6f1733309)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/sysfs.h b/include/linux/sysfs.h
index 9083628..fc5fd30 100644
--- a/include/linux/sysfs.h
+++ b/include/linux/sysfs.h
@@ -80,6 +80,11 @@ struct attribute_group {
  .show = _name##_show,      \
 }
 
+#define __ATTR_WO(_name) {      \
+ .attr = { .name = __stringify(_name), .mode = S_IWUSR }, \
+ .store = _name##_store,     \
+}
+
 #define __ATTR_RW(_name) __ATTR(_name, (S_IWUSR | S_IRUGO),  \
     _name##_show, _name##_store)
 
-- 
1.7.1